### PR TITLE
daemon: logger: error out on daemon start if invalid logger address

### PR DESCRIPTION
**Pull request by runcom**
If an invalid logger address is provided on daemon start it will
silently fail. As syslog driver is doing, this check should be done on
daemon start and prevent it from starting even in other drivers.
This patch also adds integration tests for this behavior.

Also fix wrong comment referring to old job mechanism (didn't want to open another PR just for this so it's another commit)

Signed-off-by: Antonio Murdaca runcom@linux.com

### CODE REVIEW

**Comment by runcom** [comment]
ping @LK4D4

File: api/server/container.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (s *Server) getContainersLogs(ctx context.Context, w http.ResponseWriter, r
 		return fmt.Errorf("Missing parameter")
 	}
 
-	// Validate args here, because we can't return not StatusOK after job.Run() call
+	// Validate args here, because we can't return not StatusOK after stream starts
```
Comment (cpuguy83) [inline api/server/container.go]: While you are at it, can you clarify this comment a it? `can't return not StatusOK` is super confusing.

Comment (runcom) [inline api/server/container.go]: updated, I hope it's clear, otherwise suggestions accepted :)

**Comment by calavera** [comment]
LGTM

**Comment by ehazlett** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**